### DISCUSSION TITLE
Droid: Exception When Using XamlReader on Xaml with CustomResource

### DISCUSSION THREAD

**Discussion by seanocali** [unoplatform/uno#9580] [Q&A] [answered]
> **System.InvalidOperationException:** 'Unable to find type {http://schemas.microsoft.com/winfx/2006/xaml/presentation}CustomResource'

It works fine with UWP. Is this not supported?

**Comment by jeromelaban** [comment, marked as answer]
It is indeed not supported: https://github.com/unoplatform/uno/issues/749. Which scenario do you need it for?

**Comment by seanocali** [comment]
Oh I thought it was, it says it is here:
[https://newreleases.io/project/github/unoplatform/uno/release/3.0.5](https://newreleases.io/project/github/unoplatform/uno/release/3.0.5)

My scenario is that I have a few global properties that are used in dozens of different views. It saves me from having to put references to them in dozens of viewmodels. So it's not a big deal.

**Reply by jeromelaban** [reply]
This feature is actually supported in the XAML [directly in your project](https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UI/UI/Xaml/Resources/CustomXamlResourceLoader.cs), but not in `XamlReader`. Do you need this specifically in XamlReader because you're generating XAML ?

**Reply by seanocali** [reply]
I can easily live without CustomResource, but am I to understand StaticResrouce and ThemeResource won't work with XamlReader either?

**Reply by jeromelaban** [reply]
Both work pretty well, they are used in the [Uno Playground](https://playground.platform.uno/#listview).

**Reply by seanocali** [reply]
Hey Jerome, turns out I actually really do need to use CustomResource. It's crucial for a pattern I'm using for my app to support skinning packages at runtime.

Anyway, I forked Uno and added support for CustomResource in XamlObjectBuilder. It was pretty simple since Binding isn't involved it's just reading the value from the dictionary and setting a DP, but I'm kind of lost on how to write a test for it. Are there existing XamlReader tests?

**Reply by jeromelaban** [reply]
Very nice! Thanks for taking a look a it. You can find unit tests here: https://github.com/unoplatform/uno/blob/master/src/Uno.UI.Tests/Windows_UI_Xaml_Markup/XamlReaderTests/Given_XamlReader.cs

You can open the [Uno.UI-UnitTests-only.slnf](https://github.com/unoplatform/uno/blob/master/src/Uno.UI-UnitTests-only.slnf) filter, with the `net461` override to run the tests.
